monai/networks/trt_compiler.py

```diff
@@ -0,0 +1,545 @@
+            engine_bytes = torch_tensorrt.convert_method_to_trt_engine(
+                ir_model,
+                "forward",
+                inputs=inputs,
+                ir="torchscript",
+                enabled_precisions=enabled_precisions,
+                **export_args,
+            )
+        else:
+            if self.method == "onnx_dynamo":
+                import torch_onnx
+
+                torch_onnx.patch_torch()
+                export_args["dynamo"] = True
+
+            # Use temporary directory for easy cleanup in case of external weights
+            with tempfile.TemporaryDirectory() as tmpdir:
+                onnx_path = Path(tmpdir) / "model.onnx"
+                self.logger.info(
+                    f"Exporting to {onnx_path}:\n\toutput_names={self.output_names}\n\texport args: {export_args}"
+                )
+                convert_to_onnx(
+                    model,
+                    input_example,
+                    filename=str(onnx_path),
+                    input_names=self.input_names,
+                    output_names=self.output_names,
+                    **export_args,
+                )
+                self.logger.info("Export to ONNX successful.")
+                engine_bytes = self._onnx_to_trt(str(onnx_path))
+
+        open(self.plan_path, "wb").write(engine_bytes)
+
+
+def trt_forward(self, *argv, **kwargs):
+    """
+    Patch function to replace original model's forward() with.
+    Redirects to TrtCompiler.forward()
+    """
+    return self._trt_compiler.forward(self, argv, kwargs)
+
+
+def trt_compile(
+    model: torch.nn.Module,
+    base_path: str,
+    args: Dict[str, Any] | None = None,
+    submodule: Union[str, List[str]] | None = None,
+    logger: Any | None = None,
+) -> torch.nn.Module:
+    """
+    Instruments model or submodule with TrtCompiler and replaces its forward() with TRT hook.
+    Args:
+      model: module to patch with TrtCompiler().
+      base_path: TRT plan(s) saved to "base_path[.submodule].plan" path.
+                 If base_path points to existing file (e.g. associated checkpoint),
+                 that file also becomes dependency - its mtime is added to args["timestamp"].
+      args: dict : unpacked and passed to TrtCompiler() - see TrtCompiler above for details.
+      submodule : Hierarchical id(s) of submodule to patch, e.g. ['image_decoder.decoder']
+                  If None, TrtCompiler patch is applied to the whole model.
+                  Otherwise, submodule (or list of) is being patched.
+    Returns:
+      Always returns same model passed in as argument. This is for ease of use in configs.
+    """
+
+    default_args: Dict[str, Any] = {
+        "method": "onnx",
+        "precision": "fp16",
+        "build_args": {"builder_optimization_level": 5, "precision_constraints": "obey"},
+    }
+
+    default_args.update(args or {})
+    args = default_args
+
+    if trt_imported and polygraphy_imported and torch.cuda.is_available():
```

Comment:
can we add some warning messages if not meet these conditions? Otherwise, users may not realize it.